core/store/src/flat/storage.rs

```diff
@@ -347,21 +386,71 @@ fn missing_delta_error(block_hash: &CryptoHash) -> FlatStorageError {
     FlatStorageError::StorageInternalError(format!("delta does not exist for block {block_hash}"))
 }
 
+// Determine a block to make the new flat head.
+// If `strict`, uses `block_hash` as the new flat head.
+// If not `strict`, uses the second most recent block with flat state changes from `block_hash`.
```

Comment:
```suggestion
// If not `strict`, uses the second most recent block with flat state changes from `block_hash`, if it exists.
```
or smth else to say that if there is no such block, this function is no-op